src/utils.js

```diff
@@ -236,15 +236,50 @@ function camelcase (dashCaseText) {
     })
 }
 
+export function isAppleSilicon () {
+    const canvas = document.createElement('canvas')
+    const gl = canvas.getContext('webgl')
+    const debugInfo = gl.getExtension('WEBGL_debug_renderer_info')
+    const renderer = gl.getParameter(debugInfo.UNMASKED_RENDERER_WEBGL)
+    return renderer.includes('Apple M1') || renderer.includes('Apple GPU')
+}
+
+/**
+ * Get the value of a config setting.
+ * If the value is not set, return the default value.
+ * If the value is not an object, return the value.
+ * If the value is an object, check its type property.
+ *
+ * @param {any} configSetting - The config setting to get the value for
+ * @param {any} defaultValue - The default value to use if the config setting is not set
+ * @returns The value of the config setting or the default value
+ */
+function getFeatureAttr (configSetting, defaultValue) {
+    if (configSetting === undefined) {
+        return defaultValue
+    }
+
+    if (typeof configSetting !== 'object' || !configSetting.type) {
+        return configSetting
+    }
+
+    if (configSetting.type === 'undefined') {
+        return undefined
+    } else {
+        return configSetting.value
+    }
+}
+
 /**
  * @param {string} featureName
  * @param {object} args
  * @param {string} prop
  * @returns {any}
  */
-export function getFeatureSetting (featureName, args, prop) {
+export function getFeatureSetting (featureName, args, prop, defaultValue) {
     const camelFeatureName = camelcase(featureName)
-    return args.featureSettings?.[camelFeatureName]?.[prop]
+    const configSetting = args.featureSettings?.[camelFeatureName]?.[prop]
+    return getFeatureAttr(configSetting, defaultValue)
```

Comment:
This is wrapped the wrong way around to what I meant (sorry if this wasn't clear).

_getFeatureAttr_ should call into _getFeatureSetting_ which is already in use, the return value of the _getFeatureSetting_ function should just return the raw config setting value and shouldn't be modified in this change.

src/features/fingerprinting-screen-size.js and src/features/fingerprinting-hardware.js should call into _getFeatureAttr_  instead of _getFeatureSetting_ 

_getFeatureAttr_  should handle unwrapping the config value after calling _getFeatureSetting_ 

Additionally  _getFeatureAttr_  probably should handle the AppleSilicon forking logic rather than having a special case for concurrency specifically. for example hardwareConcurrency currently doesn't support _{type: 'undefined'}_ and I think this could be handled by expressing this in the config return value.

For example the config could have:
```
"settings": {
    "keyboard": {type: 'undefined'},
    "availLeft": {type: 'number', value: 12},
    "hardwareConcurrency": [
        {type: 'number', value: 12},
        {type: 'number', value: 12, criteria: {arch: 'AppleSilicon'}},
     ]
 }
 ```
This should allow for different criteria later and also allow us to change for example 'keyboard' as a remote config update.

Happy to discuss over MM or Asana if that's more helpful than GH comments :)